### PR TITLE
[tune] Kimi-K2.5/K2.6 fp4: port BM16 (flydsl_mxmoe_g*_a4w4) for decode gemm2 (+3~13% throughput)

### PR DESCRIPTION
**Pull request by jiacao-amd**
## Summary

Switches the Kimi-K2.5/K2.6 fp4 decode token buckets (`inter_dim=512`, `expert=384`, `topk=8`) to the **port BM16 kernel path** (`flydsl_mxmoe_g{1,2}_a4w4_16x256x256`), covering tokens 1-256.

### Kernel benchmark (MI355X / gfx950, `cu_num=256`, `model_dim=7168`, `inter_dim=512`, `expert=384`, `topk=8`)

Baseline `21ae71991` vs this PR. Median of interleaved repeated runs.

| token | baseline (us) | this PR (us) | delta |
|:-----:|--------------:|-------------:|------:|
| 1   |  38.79 |  31.25 | **+19.4%** |
| 2   |  42.47 |  35.96 | **+15.3%** |
| 4   |  62.26 |  49.18 | **+21.0%** |
| 8   |  86.87 |  67.84 | **+21.9%** |
| 16  | 134.95 | 116.03 | **+14.0%** |
| 32  | 205.39 | 186.36 |  **+9.3%** |
| 64  | 308.84 | 285.08 |  **+7.7%** |
| 128 | 364.95 | 343.54 |  **+5.9%** |
| 256 | 390.64 | 369.83 |  **+5.3%** |

Every decode bucket improves. Tokens >=512 are unchanged by this PR.

Configs were selected per bucket from interleaved repeated measurement.

### Validation

`op_tests/test_moe_2stage.py --no-legacy -t 1 2 4 8 16 32 64 128 256 512 1024 2048`

- **73 cases ran, 0 AssertionError.** Both `logits_diff` warnings (max 0.0020) are 5x under the 0.01 assert threshold.

### CODE REVIEW

**Comment by Fangzhou-Ai** [comment]
This PR looks good to me; it has been confirmed earlier in other models that one of the major high conc regression in vLLM is due to the lack of BM16 MoE kernel config.

**Comment by jiacao-amd** [comment]
## Accuracy: GSM8K Full Evaluation (1319 samples)

**Setup:** port BM16 CSV, `vllm/vllm-openai-rocm:nightly`, Kimi-K2.6-MXFP4, 8-shot

| Filter | Score | Stderr |
|--------|------:|-------:|
| flexible-extract | **0.9333** | ±0.0069 |
| strict-match     | **0.9325** | ±0.0069 |

**Comment by Fangzhou-Ai** [comment]
> ## Accuracy: GSM8K Full Evaluation (1319 samples)
> **Setup:** port BM16 CSV, `vllm/vllm-openai-rocm:nightly`, Kimi-K2.6-MXFP4, 8-shot
> 
> Filter	Score	Stderr
> flexible-extract	**0.9333**	±0.0069
> strict-match	**0.9325**	±0.0069

can you compare with the baseline? 0.93 is a little bit low.

**Comment by jiacao-amd** [comment]
@Fangzhou-Ai 
| | flexible-extract | strict-match |
|---|---:|---:|
| Baseline (BM32) | 0.9318 ±0.0069 | 0.9310 ±0.0070 |
| BM16 | 0.9333 ±0.0069 | 0.9325 ±0.0069 |
| diff | +0.0015 | +0.0015 |

**Comment by valarLip** [comment]
please have a AITER UT results for these changes

**Comment by jiacao-amd** [comment]
@valarLip 
All Standard Tests pass on the latest commit (`04076758f`)
Full CI: https://github.com/ROCm/aiter/pull/4307/checks

**Review by valarLip — APPROVED** [review]
